dask/array/fft.py

```diff
@@ -50,31 +30,37 @@ def _fft_out_chunks(a, n, axis):
     return chunks
 
 
-def _rfft_out_chunks(a, n, axis):
+def _rfft_out_chunks(a, n, axis, rtype):
     if n is None:
         n = a.chunks[axis][0]
     chunks = list(a.chunks)
-    chunks[axis] = (n // 2 + 1,)
+    if issubclass(np.dtype(rtype).type, numbers.Real):
+        chunks[axis] = (n,)
+    else:
+        chunks[axis] = (n // 2 + 1,)
     return chunks
 
 
-def _irfft_out_chunks(a, n, axis):
+def _irfft_out_chunks(a, n, axis, rtype):
     if n is None:
-        n = 2 * (a.chunks[axis][0] - 1)
+        n = a.chunks[axis][0]
     chunks = list(a.chunks)
-    chunks[axis] = (n,)
+    if issubclass(a.dtype.type, numbers.Real):
+        chunks[axis] = (n,)
+    else:
+        chunks[axis] = (2 * (n - 1),)
```

Comment:
Though if there was one that felt funny, this might be it. We are relying on the type to be the correct type for the `irfft` function we are working with. So if the input array is real, we assume we are dealing with SciPy's `irfft`. However, if the input array is complex, we are assuming we are dealing with NumPy's `irfft`. However, NumPy's `irfft` can take real values in addition to complex ones; whereas, SciPy's `irfft` can only take real values. I'm not sure if I like it much. It might be ok if documented, but it does allow the user to make a potentially painful mistake.